test/e2e/certificates_test.go

```diff
@@ -354,6 +355,65 @@ var _ = Describe("ACME Certificate", Ordered, func() {
 			Expect(err).NotTo(HaveOccurred())
 		})
 	})
+
+	Context("ACME Certificate dns-01 challenge using ambient credentials should obtain a valid LetsEncrypt certificate in GCP mint/passthrough clusters-65035", func() {
+		It("should obtain a valid LetsEncrypt certificate", func() {
+
+			By("getting Infrastructure object")
+			infra, err := configClient.Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			if infra.Status.PlatformStatus.Type != configv1.GCPPlatformType {
+				Skip("Skip this case for current plaform, it's not gcp.")
+			}
+
+			By("Check cloud credential in cluster")
+			_, err = loader.KubeClient.CoreV1().Secrets("kube-system").Get(ctx, "gcp-credentials", metav1.GetOptions{})
+			if err != nil {			
+				Skip("Skipping for the cluster without credential in cluster")
+			}
+
+			By("creating CredentialsRequest object")
+			loader.CreateFromFile(testassets.ReadFile, filepath.Join("testdata", "credentials", "credentialsrequest_gcp.yaml"), "")
+
+			By("Configure cert-manager to use credential, setting this credential secret name in subscription object")
+			credentialSecret := "gcp-credentials"
+			err = patchSubscriptionWithCloudCredential(ctx, loader, credentialSecret)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating new certificate ClusterIssuer")
+			clusterIssuerName := "acme-dns01-clouddns-ambient"
+			loader.CreateFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "clusterissuer_gcp.yaml"), "")
+			defer certmanagerClient.CertmanagerV1().ClusterIssuers().Delete(ctx, clusterIssuerName, metav1.DeleteOptions{})
+
+			By("creating a test namespace")
+			ns, err := loader.CreateTestingNS("e2e-acme-ambient-dns01")
```

Comment:
This seems copied from Dev's existing scenario. It should change to use a different name for your scenario. (Did not notice this. Today found it)

Reply:
updated.